### PR TITLE
Update LeadListItem UI (progress)

### PR DESCRIPTION
**Pull request by tnagorra**
- add isFetchable check for mock response condition as well
- mock connectors response
- mock connector leads response
- separate UI for connector-lead and lead mode

### CODE REVIEW

File: src/views/LeadAdd/index.js [hunk 1]
```diff
@@ -486,25 +516,127 @@ function LeadAdd(props) {
...
     const activeLeadState = activeLeadKey
         ? leadStates[activeLeadKey]
         : undefined;
-
     const leadPreviewMinimized = leadPreviewHidden || leadIsTextType;
 
     // TODO: IMP calculate this value
     const saveEnabledForAll = false;
 
+    const leadConnectorChildren = connectors && connectors.length > 0 && (
```
Comment (AdityaKhatri) [inline src/views/LeadAdd/index.js]: Lets make this a separate component

File: src/views/LeadAdd/index.js [hunk 2]
```diff
@@ -553,71 +677,112 @@ function LeadAdd(props) {
                                             className={styles.leadButtons}
                                             onLeadsAdd={handleLeadsAdd}
                                             leadStates={leadStates}
+                                            activeSource={activeSource}
+                                            onSourceChange={(source) => {
```
Comment (AdityaKhatri) [inline src/views/LeadAdd/index.js]: use useCallback

File: src/views/LeadAdd/index.js (hunk 1 above)
Comment (tnagorra) [reply src/views/LeadAdd/index.js]: Was working on it.

File: src/views/LeadAdd/ConnectorLeadList/index.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        },
+        [activeLeadKey, onLeadSelect],
+    );
+
+    return (
+        <div className={_cs(styles.leadListContainer, className)}>
+            <ListView
+                className={styles.leadList}
+                data={leads}
+                keySelector={item => item.id}
```
Comment (AdityaKhatri) [inline src/views/LeadAdd/ConnectorLeadList/index.js:84]: define the function elsewhere

**Pull request merged**